homeassistant/components/p1_monitor/sensor.py

```diff
@@ -0,0 +1,281 @@
+        ),
+        SensorEntityDescription(
+            key="power_produced_phase_l1",
+            name="Power Produced Phase L1",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="power_produced_phase_l2",
+            name="Power Produced Phase L2",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="power_produced_phase_l3",
+            name="Power Produced Phase L3",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+    ),
+    SERVICE_SETTINGS: (
+        SensorEntityDescription(
+            key="gas_consumption_tariff",
+            name="Gas Consumption - Tariff",
+            entity_registry_enabled_default=False,
+            device_class=DEVICE_CLASS_MONETARY,
+            native_unit_of_measurement=CURRENCY_EURO,
+        ),
+        SensorEntityDescription(
+            key="energy_consumption_low_tariff",
+            name="Energy Consumption - Low Tariff",
+            device_class=DEVICE_CLASS_MONETARY,
+            native_unit_of_measurement=CURRENCY_EURO,
+        ),
+        SensorEntityDescription(
+            key="energy_consumption_high_tariff",
+            name="Energy Consumption - High Tariff",
+            device_class=DEVICE_CLASS_MONETARY,
+            native_unit_of_measurement=CURRENCY_EURO,
+        ),
+        SensorEntityDescription(
+            key="energy_production_low_tariff",
+            name="Energy Production - Low Tariff",
+            device_class=DEVICE_CLASS_MONETARY,
+            native_unit_of_measurement=CURRENCY_EURO,
+        ),
+        SensorEntityDescription(
+            key="energy_production_high_tariff",
+            name="Energy Production - High Tariff",
+            device_class=DEVICE_CLASS_MONETARY,
+            native_unit_of_measurement=CURRENCY_EURO,
+        ),
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up P1 Monitor Sensors based on a config entry."""
+    async_add_entities(
+        P1MonitorSensorEntity(
+            coordinator=hass.data[DOMAIN][entry.entry_id],
+            entry_id=entry.entry_id,
+            description=description,
+            service_key=service_key,
+            name=entry.title,
+            service=SERVICES[service_key],
+        )
+        for service_key, service_sensors in SENSORS.items()
+        for description in service_sensors
+    )
+
+
+class P1MonitorSensorEntity(CoordinatorEntity, SensorEntity):
+    """Defines an P1 Monitor sensor."""
+
+    def __init__(
```

Comment:
```suggestion
class P1MonitorSensorEntity(CoordinatorEntity, SensorEntity):
    """Defines an P1 Monitor sensor."""
    
    coordinator: P1MonitorDataUpdateCoordinator

    def __init__(
```

For the typing to be complete you should add this as well

Reply:
Why not adding this directly inside the init?

Reply:
I was wrong anyway :+1: